filetests/isa/x86/optimized-zero-constants-32bit.clif

```diff
@@ -1,5 +1,6 @@
-; Check that floating-point constants equal to zero are optimized correctly.
+; Check that floating-point and integer constants equal to zero are optimized correctly.
 test binemit
+set opt_level=best
```

Comment:
nit: Instead of changing existing tests, can you add new files instead, please? It's cheap and doesn't require modifying the behavior of current tests, which might lower coverage. (Ditto for the other file.)

Reply:
Also, if you want to ensure the code actually runs as a part of the CI you can add `test run` at the top of those tests and `; run` after any functions you want executed on the current machine. The one limitation now is that the function must return a boolean indicating the test's success or failure. Example here: https://github.com/CraneStation/cranelift/blob/master/filetests/isa/x86/binary64-run.clif.

Reply:
I'd argue that the existing tests not having `opt_level=best` is incorrect - the test file name _is_ "optimized". I think there's a read that the `pxor r, r` rule for floating point 0's has too high a precedence, since elsewhere we pick a naive but more flexible encoding first for quick unoptimized codegen (like iconst's zero-extending i64 `movl` rule currently).

How's it sound to move the existing pxor tests to another file ("floating-point-zero-constants.clif"?) and keep these under `opt_level=best`?

Reply:
(ended up moving existing pxor tests to a "floating-point-zero-constants" which is an exact copy of the "optimized-zero-constants" test on master. same for the 32-bit versions.)

Reply:
Nice, thank you!